NetCoreRestApi/DataLayer/UnitOfWorks/BaseUnitOfWork.cs

```diff
@@ -1,51 +1,66 @@
-﻿using DataLayer.Models;
+﻿using DataLayer.interfaces;
+using DataLayer.Models;
 using DataLayer.Repositories;
-using DataLayer.Repositories.Intefaces;
+using System;
 using System.Collections.Generic;
+using System.Dynamic;
 using System.Linq;
 
 namespace DataLayer.UnitOfWorks
 {
     public abstract class BaseUnitOfWork<TModel, TId> : IUnitOfWork<TModel, TId>
         where TModel : BaseModel<TId>
     {
-        protected readonly IRepository<TModel, TId> _repository;
-        protected readonly IUnitOfWorkContext _dbContext;
+        protected IRepository<TModel, TId> Repository { get; }
+        protected ITransactionManager TransactionManager { get; }
 
-        public BaseUnitOfWork(IUnitOfWorkContext dbContext, IRepository<TModel, TId> repository)
+        protected BaseUnitOfWork(ITransactionManager transactionManager, IRepository<TModel, TId> repository)
         {
-            _dbContext = dbContext;
-            _repository = repository;
+            TransactionManager = transactionManager;
+            Repository = repository;
         }
 
         public virtual TModel GetById(TId id)
         {
-            return _repository.GetById(id);
+            return Repository.GetById(id);
         }
 
-        public virtual IQueryable<TModel> GetAll()
+        public virtual IQueryable<TModel> GetAll(FilterParameters filter)
         {
-            return _repository.GetAll();
+            return Repository.GetAll(filter);
         }
 
         public virtual IEnumerable<TModel> Create(IEnumerable<TModel> models)
         {
-            return _repository.Create(models);
+            TransactionManager.BeginTransaction();
```

Comment:
Add logic for checking if BeginTransaction has been started

Reply:
done